assignments/array-methods.js

```diff
@@ -56,28 +56,67 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(runners) {
+    //fullName.push(runners['first_name'] + ' ' + runners['last_name']);
+    fullName.push(`${runners['first_name']} ${runners['last_name']}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+allCaps = runners.map(function(runners) {
+    return runners.first_name.toLowerCase();
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
-console.log(largeShirts);
+largeShirts = runners.filter(function(runners) {
+    if(runners.shirt_size === "L") {
+        return runners.shirt_size;
+    }
+});
+console.log(JSON.stringify(largeShirts));
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal = runners.reduce(function(total, runners) {
+    return total + runners.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Collect all participants emails into a single array.
+
+let contestants = [];
+contestants = runners.reduce(function(total, runners) {
+    total.push(runners.email);
+    return total;
+}, []);
+console.log(JSON.stringify(contestants));
 
 // Problem 2
+// Gather all company names into an array.
+
+let companyName = [];
+runners.forEach(function(runners) {
+    companyName.push(`${runners['company_name']}`);
+});
+console.log(companyName);
+
+// Problem 3
+// Gather info on the top donators.
 
-// Problem 3
\ No newline at end of file
+let highestDonations = [];
+highestDonations = runners.filter(function(runners) {
+    if(runners.donation > 290) {
+        return runners.donation;
```

Comment:
you can get rid of the if block, and just have `return runners.donation > 290` on line 118